From 89f01dab29d3c9a833c9c36a8f2151b6f97696c7 Mon Sep 17 00:00:00 2001
From: Laurent Vivier <lvivier@redhat.com>
Date: Fri, 24 May 2019 12:13:04 +0100
Subject: [PATCH 4/4] redhat: define pseries-rhel8.1.0 machine type

RH-Author: Laurent Vivier <lvivier@redhat.com>
Message-id: <20190524121304.15530-5-lvivier@redhat.com>
Patchwork-id: 88203
O-Subject: [RHEL-AV qemu-kvm PATCH v4 4/4] redhat: define pseries-rhel8.1.0 machine type
Bugzilla: 1709726
RH-Acked-by: David Gibson <dgibson@redhat.com>
RH-Acked-by: Cornelia Huck <cohuck@redhat.com>
RH-Acked-by: Dr. David Alan Gilbert <dgilbert@redhat.com>

update pseries-rhel8.0.0 like pseries-3.1 except
for SPAPR_CAP_CFPC, SPAPR_CAP_SBBC, SPAPR_CAP_IBS
that have already been backported.

Signed-off-by: Laurent Vivier <lvivier@redhat.com>
Signed-off-by: Danilo C. L. de Paula <ddepaula@redhat.com>
---
 hw/ppc/spapr.c | 28 ++++++++++++++++++++++++++--
 1 file changed, 26 insertions(+), 2 deletions(-)

diff --git a/hw/ppc/spapr.c b/hw/ppc/spapr.c
index 67c9e01..8adcb97 100644
--- a/hw/ppc/spapr.c
+++ b/hw/ppc/spapr.c
@@ -4670,15 +4670,39 @@ DEFINE_SPAPR_MACHINE(2_1, "2.1", false);
 #endif
 
 /*
+ * pseries-rhel8.1.0
+ */
+
+static void spapr_machine_rhel810_class_options(MachineClass *mc)
+{
+    /* Defaults for the latest behaviour inherited from the base class */
+}
+
+DEFINE_SPAPR_MACHINE(rhel810, "rhel8.1.0", true);
+
+/*
  * pseries-rhel8.0.0
+ * like spapr_compat_3_1
+ * except SPAPR_CAP_CFPC, SPAPR_CAP_SBBC and SPAPR_CAP_IBS
+ * that have been backported to pseries-rhel8.0.0
  */
 
 static void spapr_machine_rhel800_class_options(MachineClass *mc)
 {
-    /* Defaults for the latest behaviour inherited from the base class */
+    SpaprMachineClass *smc = SPAPR_MACHINE_CLASS(mc);
+
+    spapr_machine_rhel810_class_options(mc);
+    compat_props_add(mc->compat_props, hw_compat_rhel_8_0,
+                     hw_compat_rhel_8_0_len);
+
+    mc->default_cpu_type = POWERPC_CPU_TYPE_NAME("power8_v2.0");
+    smc->update_dt_enabled = false;
+    smc->dr_phb_enabled = false;
+    smc->broken_host_serial_model = true;
+    smc->default_caps.caps[SPAPR_CAP_LARGE_DECREMENTER] = SPAPR_CAP_OFF;
 }
 
-DEFINE_SPAPR_MACHINE(rhel800, "rhel8.0.0", true);
+DEFINE_SPAPR_MACHINE(rhel800, "rhel8.0.0", false);
 
 /*
  * pseries-rhel7.6.0
-- 
1.8.3.1

